src/main/java/com/google/devtools/build/lib/bazel/rules/python/python_stub_template.txt

```diff
@@ -352,12 +377,21 @@ def ExecuteFile(python_program, main_filename, args, env, module_space,
       env=env,
       cwd=workspace
     )
-    output_filename = os.environ.get('COVERAGE_DIR') + '/pylcov.dat'
+    output_filename = os.environ.get('COVERAGE_DIR') + '/pylcov.dat.tmp'
     ret_code = subprocess.call(
       [python_program, coverage_tool, "lcov", "-o", output_filename] + args,
       env=env,
       cwd=workspace
     )
+    # Coveragepy uses absolute file names in the output.  We need to strip off
```

Comment:
An alternative may be asking coveragepy to report relative paths. This issue suggests there already is some sort of support for that, albeit incomplete: https://github.com/nedbat/coveragepy/issues/963

Reply:
That would be preferable, though I'd note that first of all "relative to what?" still applies; we'd probably still need to strip off at least the workspace name.  This approach here was mostly a "it's Friday evening and I just want to see if I can get a useful coverage report before going to bed" solution - I'm definitely not thrilled with it, but it does work.

Reply:
The problem here is that when it resolves the symlinks in a runfiles tree then it ends up getting a path in either the original source tree or `$(bazel info execution_root)/{external,bazel-out}`, none of which are are useful when made relative to the current working directory.  The problem with my hacky solution to this problem is that it will only work for one of the above roots.

From my read of the code (I haven't tried it, yet) it _does_ look like if coveragepy is [configured to use relative paths](https://github.com/nedbat/coveragepy/blob/3a83e41a3668aaf8d6d9f913c5a4506c36346152/coverage/control.py#L290) then it doesn't [resolve symlinks](https://github.com/nedbat/coveragepy/blob/3a83e41a3668aaf8d6d9f913c5a4506c36346152/coverage/files.py#L148), which would fix the problem here.  The problem is that behavior seems to only be controllable through the config file rather than a command line flag, which can be tricky to get working properly in a bazel sandbox environment.  Various solutions to that problem:
1. Generate a config file in a temp directory for each test invocation.
2. Include a config file probably as a target somewhere in `bazel_tools` so it could be shared by all test instantiations
3. Add a way for the user to add specify a config file in toolchain configuration.  This has some advantages, but is also prone to causing problems.

I don't love any of those but would probably go with the first approach.  I'll try it out on Tuesday.

Reply:
Unfortunately it seems like coveragepy's relative_paths feature doesn't actually work, as it still canonicalizes paths in the lcov report (as that bug you link pointed out), though it does at least seem to use relative paths in the intermediate data and xml output.  That said, there _is_ a better way to do what I was doing, which I'll push up here once I've tested it properly.